test/reduce_test.cpp

```diff
@@ -833,7 +833,7 @@ struct reduce_driver : test_driver
             auto rand_value = tensor_elem_gen_integer{max_value}(is...);
             auto sign_value = tensor_elem_gen_checkboard_sign{}(is...);
 
-            return (sign_value * rand_value + 1.0);
+            return (sign_value * rand_value / max_value + 0.01);
```

Comment:
Are you sure that changes in this file do not affect the quality of testing?

Reply:
MIOpenDriver uses values in `(0,1)` to initialize the input data,  the change here use similar method.   The previous codes use integers in `[-16, 17]` for ADD/AVG.  But I believe the accumulative error can only be reduced by using more fine-grained values.  Here the max_value is improved to `999` and use rand_value/max_value will cause more fine-grained values.  `+ 0.01` is used to avoid the mean of zero.  I could not work out better method to conquer that case,  I don't want to add `--tolerance` option, since the tolerance value in `test_reduce_test` is hard to understand, it is a multiplier used to multiply with the `std::numeric_limits<float>::epsilon` to get a final tolerance threshold. 

In general,  `test_reduce_test` is similar to other `test/*.cpp`,  which depend on the same facility and are different from `driver/*.cpp`